website/blog/2024-03-07-using-inner-blocks.md

```diff
@@ -0,0 +1,162 @@
+
+This approach works perfectly fine when you have a strict layout and don’t want larger content structure variations. Using inner blocks may introduce a bit more complexity - both for developers and content editors, but ultimately gives the editors much more freedom.
+
+Another example where inner blocks are used is the Carousel block. Using inner blocks allows you to add as many slides as you like in your carousel. Let’s not forget about Columns and Group blocks. These also use inner blocks.
+
+## Adding inner block support
+
+The example used in this blog post is a two-column Card that was made for one project. First thing that needs to be updated is the ***manifest.json*** file. For the sake of simplicity and not having too much code, the parts that aren’t related to the inner blocks have been removed.
+
+```json
+{
+	"hasInnerBlocks": true,
+	"innerBlocksDependency": [
+		"heading",
+		"paragraph"
+	],
+	"attributes": {
+		"cardTwoColumnAllowedBlocks": {
+			"type": "array",
+			"items": {
+				"type": "string"
+			},
+			"default": [
+				"eightshift-boilerplate/paragraph",
+				"eightshift-boilerplate/heading",
+				"eightshift-boilerplate/button",
+				"eightshift-boilerplate/lists"
+			]
+		}
+	}
+}
+```
+
+The `hasInnerBlocks` key defines that this block supports inner blocks. The `innerBlocksDependency` is not mandatory, but it can be used to add inner blocks automatically. You can see an example of using it in the Columns block `editor.js` file.
+
+In case we need to limit the use of inner blocks, we can do so by defining the allowed blocks inside attributes. The attribute name should be in the format `{blockName}AllowedBlocks`. To define the allowed blocks, simply add the full block names in the `default` key array, as shown in the example.
+
+The next step is to add inner block support to the `{block-name}-editor.js` file. Here’s an example of the code related only to the inner blocks:
+
+```jsx
+import { InnerBlocks } from '@wordpress/block-editor';
+import { BlockInserter } from '@eightshift/frontend-libs/scripts';
+
+export const CardTwoColumnEditor = ({ attributes, setAttributes, clientId }) => {
+
+	const {
+		cardTwoColumnAllowedBlocks,
+	} = attributes;
+
+	return (
+
+		<div className={contentClass}>
+			<InnerBlocks
+				allowedBlocks={(typeof cardTwoColumnAllowedBlocks === 'undefined') || cardTwoColumnAllowedBlocks}
+				orientation='vertical'
+				renderAppender={() => <BlockInserter clientId={clientId} hasLabel additionalClasses='es-mb-6' />}
+			/>
+		</div>
+
+	);
+};
+```
+
+The `InnerBlocks` element is the default one from the Gutenberg editor. In order to limit the allowed blocks, we need to pass the attribute that contains the list of allowed inner blocks. The `orientation` prop determines if the inner blocks will be displayed horizontally or vertically in the editor.
```

Comment:
```suggestion
The `InnerBlocks` component comes from Gutenberg. In order to limit the allowed blocks, we need to pass the the list of allowed inner blocks to the `allowedBlocks` attribute. If your inner blocks should be laid out horizontally in the editor, instead of vertically, you can pass `orientation='horizontal'`.
```